README.md

```diff
@@ -20,65 +20,50 @@ First, install the library:
 go get github.com/cloudchacho/taskhawk-go
 ```
 
-Convert your function into a "Task" as shown here:
+If your function takes multiple arguments, convert your function into a "Task" as shown here:
 
 ```go
 type SendEmailTaskInput struct {...}
 
-type SendEmailTask struct {
-    taskhawk.Task
-}
-
-func (t *SendEmailTask) Run(context context.Context, rawInput interface{}) error {
-    input := rawInput.(*SendEmailTaskInput)
+func SendEmail(ctx context.Context, input *SendEmailTaskInput) error {
     // send email
 }
 ```
 
-Tasks may accept input of arbitrary type as long as it's serializable to JSON
+Tasks may accept input of arbitrary pointer type as long as it's serializable to JSON. Remember to export fields!
 
-Then, define a few required settings:
+Then, define your backend:
 
 ```go
-sessionCache := NewAWSSessionsCache()
-
-settings := taskhawk.Settings{
+settings := aws.Settings{
     AWSAccessKey: <YOUR AWS ACCESS KEY>,
     AWSAccountID: <YOUR AWS ACCOUNT ID>,
     AWSRegion: <YOUR AWS REGION>,
     AWSSecretKey: <YOUR AWS SECRET KEY>,
 
     Queue: <YOUR TASKHAWK QUEUE>,
 }
-taskhawk.InitSettings(settings)
+backend := aws.NewBackend(settings, nil)
 ```
 
 Before the task can be dispatched, it would need to be registered, as shown below.
-It is recommended that the task names are centrally managed by the application.
 
 ```go
-taskRegistry, _ := NewTaskRegistry(NewPublisher(sessionCache, settings))
-taskRegistry.RegisterTask(&SendEmailTask{
-    Task: taskhawk.Task{
-        Inputer: func() interface{} {
-            return &SendEmailTaskInput{}
-        },
-        TaskName: "SendEmailTask",
-    },
-})
+hub := NewHub(Config{...}, backend)
```

Comment:
This is still sort of the task registry right? Whats the reason for the name change?

Reply:
It is functionally the same thing, yes. Two reasons for the change: 1. Registry seemed an odd choice of naming for a struct that's the primary way to interact with this library (e.g. it provides methods for starting the consumer, in addition to registering tasks so it's more than a "registry"), and 2. Shorter name without losing meaning.

Reply:
ah I see makes sense